src/discordcommandhandler.ts

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2019 matrix-appservice-discord
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { DiscordBot } from "./bot";
+import * as Discord from "discord.js";
+import { Util, ICommandActions, ICommandParameters, CommandPermissonCheck } from "./util";
+import { Bridge } from "matrix-appservice-bridge";
+export class DiscordCommandHandler {
+    constructor(
+        private bridge: Bridge,
+        private discord: DiscordBot,
+    ) { }
+
+    public async Process(msg: Discord.Message) {
+        const chan = msg.channel as Discord.TextChannel;
+        if (!chan.guild) {
+            await msg.channel.send("**ERROR:** only available for guild channels");
+            return;
+        }
+
+        const intent = this.bridge.getIntent();
+
+        const actions: ICommandActions = {
+            approve: {
+                description: "Approve a pending bridge request",
+                params: [],
+                permission: "MANAGE_WEBHOOKS",
+                run: async () => {
+                    if (await this.discord.Provisioner.MarkApproved(chan, msg.member, true)) {
+                        return "Thanks for your response! The matrix bridge has been approved";
+                    } else {
+                        return "Thanks for your response, however" +
+                            "the time for responses has expired - sorry!";
+                    }
+                },
+            },
+            ban: {
+                description: "Bans a user on the matrix side",
+                params: ["name"],
+                permission: "BAN_MEMBERS",
+                run: this.ModerationActionGenerator(chan, "ban", "Banned"),
```

Comment:
Do we need "ban", "Banned" as parameters to `ModerationActionGenerator`. Can't the function determine "ban" == "Banned"?

Reply:
doesn't work for `kick` -> `Kicked` and `unban` -> `Unbanned` because language is....weird